repose-aggregator/core/repose-core/src/main/java/org/openrepose/core/services/reporting/metrics/MetricsJmxObjectNameFactory.java

```diff
@@ -23,12 +23,10 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.inject.Named;
 import javax.management.MalformedObjectNameException;
 import javax.management.ObjectName;
 import java.util.Hashtable;
 
-@Named
```

Comment:
So the problem with wiring this in is that this package is not scanned in certain unit tests. The `CoreSpringProvider` determines which package to scan based on a configuration file, as you can see here:

https://github.com/rackerlabs/repose/blob/master/repose-aggregator/core/repose-core/src/main/java/org/openrepose/core/spring/CoreSpringProvider.java#L49

Well, in some of our tests, we configure the `CoreSpringProvider` to only look in a test package rather than the standard package:
https://github.com/rackerlabs/repose/blob/master/repose-aggregator/core/repose-core/src/test/resources/springConfiguration.conf#L1

As a result, the annotated `MetricsJmxObjectNameFactory` is never registered as a bean, as so it can't be found when Spring tries to wire it into the naming strategy and metrics service beans.

So we could make this an annotated bean, but it would involving changing the semantics, or at least the behavior, of those tests. I would have preferred to use this as an annotated bean, but didn't want to scope creep. If you prefer that, though, I can.